lib/charms/mongodb_libs/v0/relations.py

```diff
@@ -0,0 +1,171 @@
+                    continue
+                logger.info("Create relation user: %s on %s", config.username, config.database)
+                mongo.create_user(config)
+                self._set_relation(config)
+
+            for username in relation_users.intersection(database_users):
+                config = self._get_config(username)
+                logger.info("Update relation user: %s on %s", config.username, config.database)
+                mongo.update_user(config)
+
+            if not self.charm.model.config["auto-delete"]:
+                return
+
+            database_dbs = mongo.get_databases()
+            relation_dbs = self._get_databases_from_relations(departed_relation_id)
+            for database in database_dbs - relation_dbs:
+                logger.info("Drop database: %s", database)
+                mongo.drop_database(database)
+
+    def _get_config(self, username: str) -> MongoDBConfiguration:
+        """Construct config object for future user creation."""
+        relation = self._get_relation(username)
+        return MongoDBConfiguration(
+            replset=self.charm.app.name,
+            database=self._get_relation_database(relation),
+            username=username,
+            password=generate_password(),
+            hosts=self.charm.mongodb_config.hosts,
+            roles=self._get_relation_roles(relation),
+        )
+
+    def _set_relation(self, config: MongoDBConfiguration):
+        """Save all output fields into application relation."""
+        relation = self._get_relation(config.username)
+        if relation is None:
+            return None
+        relation.data[self.charm.app]["username"] = config.username
+        relation.data[self.charm.app]["password"] = config.password
+        relation.data[self.charm.app]["database"] = config.password
+        relation.data[self.charm.app]["endpoints"] = ",".join(config.hosts)
+        relation.data[self.charm.app]["replset"] = config.replset
+        relation.data[self.charm.app]["uris"] = config.uri
+
+    @staticmethod
+    def _get_username(relation_id: str) -> str:
+        """Construct username."""
+        return f"relation-{relation_id}"
+
+    def _get_users_from_relations(self, departed_relation_id: Optional[int]):
+        """Return usernames for all relations."""
+        relations = self.model.relations[REL_NAME]
+        return set([
+            self._get_username(relation.id)
+            for relation in relations
+            if relation.id != departed_relation_id
+        ])
+
+    def _get_databases_from_relations(self, departed_relation_id: Optional[int]) -> Set[str]:
+        """Return database names from all relations."""
+        relations = self.model.relations[REL_NAME]
+        return set([
+            self._get_relation_database(relation)
+            for relation in relations
+            if relation.id != departed_relation_id
+        ])
+
+    def _get_relation(self, username: str) -> Relation:
```

Comment:
```suggestion
    def _get_relation_from_username(self, username: str) -> Relation:
```
follow naming scheme

Reply:
fixed in https://github.com/delgod/mongodb-k8s-operator/pull/3/commits/847a2ed26d6e9b11a2606b414e108ed8650f447d